x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/expression/literal/IntervalUtilsTests.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.sql.expression.literal;
+
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.sql.expression.literal.IntervalUtils.TimeUnit;
+import org.elasticsearch.xpack.sql.parser.ParsingException;
+
+import java.time.Duration;
+import java.time.Period;
+import java.time.temporal.TemporalAmount;
+import java.util.Locale;
+
+import static java.lang.String.format;
+import static org.elasticsearch.xpack.sql.expression.literal.IntervalUtils.intervalType;
+import static org.elasticsearch.xpack.sql.expression.literal.IntervalUtils.of;
+import static org.elasticsearch.xpack.sql.expression.literal.IntervalUtils.parseInterval;
+import static org.elasticsearch.xpack.sql.tree.Location.EMPTY;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_DAY;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_DAY_TO_HOUR;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_DAY_TO_MINUTE;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_DAY_TO_SECOND;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_HOUR;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_HOUR_TO_MINUTE;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_HOUR_TO_SECOND;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_MINUTE;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_MINUTE_TO_SECOND;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_MONTH;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_SECOND;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_YEAR;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_YEAR_TO_MONTH;
+
+public class IntervalUtilsTests extends ESTestCase {
```

Comment:
Just a suggestion: instead of using the `randomSign()` maybe use the `@ParametersFactory` to run the methods with `+`, `-` and no prefix sign at all.

Reply:
Done.